### PR TITLE
objdump: add support for segments flags in the linking section

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: src/binary-reader-objdump.cc [hunk 1]
```diff
@@ -1840,16 +1842,50 @@ Result BinaryReaderObjdump::PrintSymbolFlags(uint32_t flags) {
...
     PrintDetails(" explicit_name");
-  if (flags & WABT_SYMBOL_FLAG_NO_STRIP)
+    flags &= ~WABT_SYMBOL_FLAG_EXPLICIT_NAME;
+  }
+  if (flags & WABT_SYMBOL_FLAG_NO_STRIP) {
     PrintDetails(" no_strip");
+    flags &= ~WABT_SYMBOL_FLAG_NO_STRIP;
+  }
+  if (flags != 0) {
+    PrintDetails(" %d", flags);
```
Comment (dschuff) [inline src/binary-reader-objdump.cc:1865]: so the idea is that this just prints leftover/unknown flags? If so, why not make it explicitly say it's unknown, as you do with segment flags?

File: test/binary/linking-section.txt [hunk 2]
```diff
@@ -23,10 +23,10 @@ section("linking") {
     count[2]
     str("data1")
     p2align[2]
-    flags[0]
+    flags[1]
```
Comment (dschuff) [inline test/binary/linking-section.txt:26]: wow is this file basically assembly for wabt?

**Review by dschuff — COMMENTED** [review]
I do like the improvement in readability too

File: test/binary/linking-section.txt (hunk 2 above)
Comment (sbc100) [reply test/binary/linking-section.txt:26]: Kind of.. more like a binaryen file constructor... kinda cool right?

File: src/binary-reader-objdump.cc (hunk 1 above)
Comment (sbc100) [reply src/binary-reader-objdump.cc:1865]: Yeah.. I agree.  Will do.

**Review by dschuff — APPROVED** [review]

**Pull request merged**